### PR TITLE
[Bugfix] Flamer's damage should be approriate for objects that handle it

### PR DESCRIPTION
**Pull request by TeDGamer**
## About The Pull Request
Instead of objects taking 0 damage, they now take damage similar to the turf. The damage is based on the flamer fuel.

These use to take their default parameter value of damage but then it was set to 0 for some reason thus I changed it to the damage based on the flamer fuel similar to turf so stronger fires burn them faster rather than a constant damage based on a magic number

Objects that should take fire damage will now take flame damage appropriately:
- Barricades. Yes, make sure not to flame your own cades
```
/obj/structure/barricade/flamer_fire_act(var/dam = BURN_LEVEL_TIER_1)

	take_damage(dam * burn_multiplier)
```
- Thickbush
 ```
/obj/structure/flora/jungle/thickbush/flamer_fire_act(var/dam = BURN_LEVEL_TIER_1)

	health -= dam

	healthcheck(src)
```
- Alien doors
```
/obj/structure/mineral_door/resin/flamer_fire_act(var/dam = BURN_LEVEL_TIER_1)

	health -= dam

	healthcheck()
```

## Why It's Good For The Game
One of the weird things I see in the game after fixing the flamers melting walls is now doors are left standing. So after an incidenary OB, you would see every xeno wall burn but the doors are strong, so strong they survive if they still have support. This never really made sense to me so I took a look at the code and found that objects have 0 damage which makes no sense imo since there's code to handle damage for certain objects like mentioned above ( doors, thickbush and barricades).

This fixes the issue where resin doors are impossible to burn down. This however introduces barricades being burned ( since it has it's own handler that takes damage ) so flamers need to be more careful not to flame their own cades or at least put out the fire if the barricade is on fire.

## Changelog

:cl: TeDGamer
fix: Flamers should deal non-zero damage to objects appropriately (resin doors, barricades, thickbush ).
/:cl:

### CODE REVIEW

**Comment by sose96** [comment]
I think it should depend on fire type. 
Basic bitch fire is not going to melt a solid steel/titanium cade, whereas OB giga fire probably would.

**Comment by TeDGamer** [comment]
> I think it should depend on fire type. Basic bitch fire is not going to melt a solid steel/titanium cade, whereas OB giga fire probably would.

I would assume that's what the burn multiplier would be for. Sadly, I'm not allowed to touch numbers. This is basically just changing what should be a 0 to an actual number like it used to be + modifying it based on the flame due to new fire changes between the cause_data time and now.

Sadly, I'm only contributing to fix the bug, not balance

**Comment by JeserTheFox** [comment]
Flamer fire still burns resin walls, just slowly, unless it's X napalm or something even stronger.

**Comment by Nanu308** [comment]
This in essence is fine for fixing an issue, but numbers may need to see some tweaks further down the line.

**Comment by TeDGamer** [comment]
> I know for a fact, that flamer fire still burns resin walls and doors, just slowly, unless it's X napalm or something even stronger. Cades also take damage from flames right now.

Yeah I made the resin wall burn change on gitlab. Resin walls are turf so that's the turf code that you see being slightly changed in this PR. That's the burn over time.

For doors, it's likely just the applied effect on the flamer on click similar to how it used to be for walls. Same for cades, just the applied effect of the flamer on hit rather than the fire tile itself damaging objects.

> This in essence is fine for fixing an issue, but numbers may need to see some tweaks further down the line.

Sounds good to me. Should be easy enough to modify the formula of any of the fire_flamer_act specifically to the object itself.

**Comment by TeDGamer** [comment]
> Flamers already damage marine barricades; unless there's something funny going on this shouldn't change anything with respect to flamers and cades as they already are.

That might on hit. This is DoT.

The PR itself is relatively simple, you can see all I'm changing is 0 damage to objects (mainly relevant to the three posted above: resin doors, barricades, thickbush) to a non-zero value / damage, similar to the turf damage in the same code ( this handles the burning of resin walls )

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/flamer/flamer.dm [hunk 1]
```diff
@@ -555,11 +555,12 @@
 
 /obj/flamer_fire/process(delta_time)
 	var/turf/T = loc
+	var/damage = burnlevel*delta_time
```
[the remark this answers is not in the document]
Comment (TeDGamer) [reply code/modules/projectiles/guns/flamer/flamer.dm]: Done!

**Review by Geevies — APPROVED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**